src/components/cards/Cards.js

```diff
@@ -1,20 +1,48 @@
+import { useEffect, useState } from 'react'
 import Card from './card/Card'
 import './Cards.css'
-import { getDataArr } from './getDataArr'
 
 const Cards = props => {
-  let dataArr = getDataArr()
-  console.log('Cards-dataArr is of type: ' + typeof dataArr)
+  const [dataArr, setDataArr] = useState(null)
+  const [loading, setLoading] = useState(true)
+  const [error, setError] = useState(null)
 
-  if (props.house !== 'all') {
-    dataArr = dataArr.filter(
-      person => person.house.toLowerCase() === props.house
-    )
+  useEffect(() => {
+    fetch('http://hp-api.herokuapp.com/api/characters')
+      .then(response => {
+        if (response.ok) {
+          return response.json()
+        }
+        throw response
+      })
+      .then(data => {
+        setDataArr(data)
+      })
+      .catch(err => {
+        console.error('Error fetching data: ' + err)
+        setError(err)
+      })
+      .finally(() => {
+        setLoading(false)
+      })
+  }, [])
+
+  if (loading) {
+    return <main className="cards">Currently loading..</main>
+  }
+
+  if (error) {
+    return <main className="cards">{error}</main>
   }
 
+  let workingData =
```

Comment:
Why is this declared with `let` and not `const`? Also `workingData` is not a good name. Usually, we want to name the things as they are or what they contain. In this case `characters`. Maybe here `filteredCharacters` would make more sense?
```suggestion
  const filteredCharacters =
```

Reply:
I actually had it a const before, but then I got an 'warning' for line 45 when I map it:

> workingData.map((person, i) => (...)

I changed the naming, I wasn't really happy about it from the getgo